pkg/segment/writer/packer.go

```diff
@@ -1477,21 +1443,26 @@ func WriteMockBlockSummary(file string, blockSums []*BlockSummary,
 	}
 }
 
-func (ss *SegStore) checkAddDictEnc(colWip *ColWip, cval []byte, recNum uint16) {
+func (ss *SegStore) checkAddDictEnc(colWip *ColWip, cval []byte, recNum uint16, cbufIdx uint32) {
```

Comment:
this param (`cbufidx`) is the startOffset of where the last columnValue was written, we cannot extract that from colWip, since colwip's cbufidx points to the start of the next new incoming columnValue. 
So in order to get to the last columnvalue, we need 2 of these 3 things : startIdx, endIdx, len.
currently we have the `endIdx` (colWip.cbufidx), we just need one of other 2